vermouth/tests/test_itp_files.py

```diff
@@ -0,0 +1,593 @@
+        assert 'GLY' in ff.blocks
+        assert ff.blocks['GLY'].nrexcl == 3
+
+    @staticmethod
+    def test_multiple_moleculetype():
+        lines = """
+            [ moleculetype ]
+            GLY  3
+            
+            [ moleculetype ]
+            VAL 2
+            """
+        lines = textwrap.dedent(lines).splitlines()
+        ff = vermouth.forcefield.ForceField(name='test_ff')
+        vermouth.gmx.itp_read.read_itp(lines, ff)
+        assert set(ff.blocks) == set(['GLY', 'VAL'])
+        assert ff.blocks['GLY'].nrexcl == 3
+        assert ff.blocks['VAL'].nrexcl == 2
+
+    @staticmethod
+    def test_atoms():
+        lines = """
+        [ moleculetype ]
+        GLY 1
+
+        [ atoms ]
+        1 P4  1  ALA  BB  1
+        2 P3\t1\tALA  SC1 2 -3 
+        3 P2  1  ALA  SC2 3 -3 72
+        """
+        lines = textwrap.dedent(lines).splitlines()
+        ff = vermouth.forcefield.ForceField(name='test_ff')
+        vermouth.gmx.itp_read.read_itp(lines, ff)
+        block = ff.blocks['GLY']
+        assert len(block.nodes) == 3
+        assert block.nodes['1'] == {'atomname': 'BB', 'atype': 'P4',
+                                     'resname': 'ALA', 'resid': 1,
+                                     'charge_group': 1}
+        assert block.nodes['2'] == {'atomname': 'SC1', 'atype': 'P3',
+                                      'resname': 'ALA', 'resid': 1,
+                                      'charge_group': 2, 'charge': -3.0}
+        assert block.nodes['3'] == {'atomname': 'SC2', 'atype': 'P2',
+                                      'resname': 'ALA', 'resid': 1,
+                                      'charge_group': 3,'charge':-3,'mass':72}
+
+    @staticmethod
+    def test_fixed_number_interaction():
+        """
+        Define an interaction for which the number of atoms required is known.
+        """
+        lines = """
+        [ moleculetype ]
+        GLY 1
+
+        [ atoms ]
+        1 P4 1 ALA BB 1
+        2 P3 1 ALA SC1 2
+        3 P2 1 ALA SC2 3
+
+        [ bonds ]
+        1 2 1 0.2 100
+        2 3 4 0.6 700
+        3 1 -- 9 plop toto
+        """
+        lines = textwrap.dedent(lines).splitlines()
+        ff = vermouth.forcefield.ForceField(name='test_ff')
+        vermouth.gmx.itp_read.read_itp(lines, ff)
+        block = ff.blocks['GLY']
+
+        bonds = [
+            vermouth.molecule.Interaction(
+                atoms=['1', '2'], parameters=['1', '0.2', '100'], meta={},
+            ),
+            vermouth.molecule.Interaction(
+                atoms=['2', '3'], parameters=['4', '0.6', '700'],meta={},
+            ),
+            vermouth.molecule.Interaction(
+                atoms=['3', '1'], parameters=['9', 'plop', 'toto'], meta={},
+            ),
+        ]
+        assert block.interactions['bonds'] == bonds
+
+
+    @staticmethod
+    def test_interaction_by_index():
+        """
+        Create an interaction and refer to atoms by index.
+        """
```

Comment:
What does this test differently than the previous?

Reply:
basically indices are used to define the atoms of interactions and within the FF context this check was checking for atom names.